frontend/apps/web/app/(mgmt)/[account]/new/connection/postgres/PostgresForm.tsx

```diff
@@ -130,6 +136,66 @@ export default function PostgresForm() {
     }
   }
 
+  /* we call the underlying useGetConnection API directly since we can't call
+the hook in the useEffect conditionally. 
+*/
+  useEffect(() => {
+    const fetchData = async () => {
+      if (sourceConnId && account) {
+        setIsLoading(true);
+        try {
+          const response = await fetch(
+            `/api/accounts/${account?.id ?? ''}/connections/${sourceConnId}`,
+            {
+              method: 'GET',
+              headers: {
+                'Content-Type': 'application/json',
+              },
+            }
+          );
+          const data: GetConnectionResponse = await response.json();
+
+          const connData =
+            data instanceof GetConnectionResponse
+              ? data
+              : GetConnectionResponse.fromJson(data);
+
+          if (connData) {
+            const config = connData.connection?.connectionConfig?.config
+              .value as PostgresConnectionConfig;
+
+            const pgConfig = config.connectionConfig
+              .value as PostgresConnection;
+
+            setIsLoading(false);
+            // reset the form with the new values
+            form.reset({
+              ...form.getValues(),
+              connectionName: pgConfig.name + '_copy',
+              db: {
+                ...form.getValues().db,
+                host: pgConfig.host,
+                name: pgConfig.name,
+                user: pgConfig.user,
+                pass: pgConfig.pass,
+                port: pgConfig.port,
+                sslMode: pgConfig.sslMode,
+              },
+            });
+          }
+        } catch (error) {
+          console.error('Failed to fetch connection data:', error);
+        }
+      }
+    };
+
+    fetchData();
+  }, [sourceConnId, account, form]);
```

Comment:
you only want this useEffect to run once, so it should not have any dependencies, just an empty `[]`.

Reply:
If you're worried about the account id not being present, then there needs to be a trigger to ensure that this fetch only happens a single time and doesn't re-run and reset the form on the user.

Also, it's generally a bad idea to use objects as a dependency unless you're 100% sure they are memoized. It's generally better to key off one of the properties that you're using instead as it's a stable value. For example you're using only the `account.id` here. The dependency should instead be `account?.id` instead of `account`.

Reply:
makes sense, updated it so that only the account.Id is in the dependency array and the page waits for the `useAccount` hook to return before rendering the page